extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeStreamExtractor.java

```diff
@@ -193,74 +193,190 @@ public String getSubChannelAvatarUrl() {
     @Nonnull
     @Override
     public String getDashMpdUrl() {
-        return "";
+        return EMPTY_STRING;
     }
 
     @Nonnull
     @Override
     public String getHlsUrl() {
-        return json.getArray("streamingPlaylists").getObject(0).getString("playlistUrl");
+        assertPageFetched();
+        return json.getObject("files").getString("playlistUrl", EMPTY_STRING);
     }
 
     @Override
-    public List<AudioStream> getAudioStreams() {
-        return Collections.emptyList();
+    public List<AudioStream> getAudioStreams() throws ParsingException {
+        assertPageFetched();
+        final List<AudioStream> audioStreams = new ArrayList<>();
+        // Non-HLS streams
+        try {
+            audioStreams.addAll(getAudioStreamsFromArray(json.getArray("files"), ""));
+        } catch (final Exception ignored) {
+        }
+        // HLS streams
+        try {
+            final JsonArray streamingPlaylists = json.getArray("streamingPlaylists");
+            for (final Object p : streamingPlaylists) {
+                if (!(p instanceof JsonObject)) continue;
+                final JsonObject playlist = (JsonObject) p;
+                final String playlistUrl = playlist.getString("playlistUrl");
+                audioStreams.addAll(getAudioStreamsFromArray(playlist.getArray("files"),
+                        playlistUrl));
+            }
+        } catch (final Exception e) {
+            throw new ParsingException("Could not get video streams", e);
```

Comment:
I don't understand this. If we get audio streams in the first try/catch block, why do we throw an exception below? And why is the message `Could not get video streams`?

Reply:
> And why is the message Could not get video streams?

Already fixed.

> If we get audio streams in the first try/catch block, why do we throw an exception below?

So should I store the exceptions which occured, if there are any, after the last catch block is reached, and should I throw the first one only if the `audioStreams`/`videoStreams` list is empty?

Reply:
We have plenty places where we ignore exceptions in the hope to get a result later. Unfortunately, we are not informed about these extracion failures if some later code is successful.
I think the proper solution is to add a list to the `Extractor`  class  which stores "secondary" exeptions. Than all methods which ignore Exceptions, because they have the posibility to get a result later, can add the caught exceptions to that list. The list could later be merged with [this `errors` list](https://github.com/TeamNewPipe/NewPipeExtractor/blob/d8179dc35ae036bfd6e450a7050e998e0f1b041f/extractor/src/main/java/org/schabi/newpipe/extractor/Info.java#L34) in the Collectors. This allows us to 
1. create tests to check if the list is empty
2. get notified by users, when they run into these extraction problems.

I think this is out of scope for this PR. If you aggree to my proposal, we should open an issue with the information from above.